aspnetcore/mvc/controllers/routing.md

```diff
@@ -167,570 +235,656 @@ public class ProductsController : Controller
+
+ See [Routing](xref:fundamentals/routing) for a full description of route templates and related options.
+
+For more information on `[ApiController]`, see [ApiController attribute](xref:web-api/index##apicontroller-attribute).
+
+## Route name
+
+The following code  defines a route name of `Products_List`:
+
+[!code-csharp[](routing/samples/3.x/main/Controllers/ProductsApiController.cs?name=snippet2)]
+
+Route names can be used to generate a URL based on a specific route. Route names:
+
+* Have no impact on the URL matching behavior of routing.
+* Are only used for URL generation.
+
+Route names must be unique application-wide.
+
+Contrast the preceding code with the conventional default route which defines the `id` parameter as optional (`{id?}`). The ability to precisely specify APIs has advantages, such as  allowing `/products` and `/products/5` to be dispatched to different actions.
 
 <a name="routing-combining-ref-label"></a>
 
-### Combining routes
+## Combining attribute routes
 
 To make attribute routing less repetitive, route attributes on the controller are combined with route attributes on the individual actions. Any route templates defined on the controller are prepended to route templates on the actions. Placing a route attribute on the controller makes **all** actions in the controller use attribute routing.
 
-```csharp
-[Route("products")]
-public class ProductsApiController : Controller
-{
-   [HttpGet]
-   public IActionResult ListProducts() { ... }
+[!code-csharp[](routing/samples/3.x/main/Controllers/ProductsApiController.cs?name=snippet)]
+
+In the preceding example:
+
+* The URL path `/products` can match `ProductsApi.ListProducts`
+* The URL path `/products/5` can match `ProductsApi.GetProduct(int)`.
+
+Both of these actions only match HTTP `GET` because they're marked with the `[HttpGet]` attribute.
+
+Route templates applied to an action that begin with `/` or `~/` don't get combined with route templates applied to the controller. The following example matches a set of URL paths similar to the default route.
+
+[!code-csharp[](routing/samples/3.x/main/Controllers/HomeController.cs?name=snippet)]
+
+The following table explains the `[Route]` attributes in the preceding code:
+
+| Attribute               | Combines with `[Route("Home")]` | Defines route template |
+| ----------------- | ------------ | --------- |
+| `[Route("")]` | Yes | `"Home"` |
+| `[Route("Index")]` | Yes | `"Home/Index"` |
+| `[Route("/")]` | **No** | `""` |
+| `[Route("About")]` | Yes | `"Home/About"` |
+
+<a name="routing-ordering-ref-label"></a>
+<a name="oar"></a>
+
+### Attribute route order
+
+<!-- In the  original, you had "In contrast to conventional routes which execute in a defined order"
```

Comment:
If there are still references to this we should remove them. Conventional routes define a "matching priority", not an "execution order"